executor/aggfuncs/func_count_distinct.go

```diff
@@ -433,3 +435,380 @@ func appendJSON(encodedBytes, _ []byte, val json.BinaryJSON) []byte {
 	encodedBytes = append(encodedBytes, val.Value...)
 	return encodedBytes
 }
+
+func intHash64(x uint64) uint64 {
+	x ^= x >> 33
+	x *= 0xff51afd7ed558ccd
+	x ^= x >> 33
+	x *= 0xc4ceb9fe1a85ec53
+	x ^= x >> 33
+	return x
+}
+
+type baseApproxCountDistinct struct {
+	baseAggFunc
+}
+
+const (
+	// The maximum degree of buffer size before the values are discarded
+	uniquesHashMaxSizeDegree uint8 = 17
+	// The maximum number of elements before the values are discarded
+	uniquesHashMaxSize = uint32(1) << (uniquesHashMaxSizeDegree - 1)
+	// Initial buffer size degree
+	uniquesHashSetInitialSizeDegree uint8 = 4
+	// The number of least significant bits used for thinning. The remaining high-order bits are used to determine the position in the hash table.
+	uniquesHashBitsForSkip = 32 - uniquesHashMaxSizeDegree
+)
+
+type approxCountDistinctHashValue uint32
+
+// Based on BJKST algorithm.
```

Comment:
The comment of a struct should be begin with `struct name ...`

Reply:
fixed